src/pages/view-test-result/cat-home-test/__tests__/view-test-result.cat-home-test.page.spec.ts

```diff
@@ -0,0 +1,277 @@
+import { BusinessDetailsCardComponent } from '../../components/business-details-card/business-details-card';
+import { ContactDetailsCardComponent } from '../../components/contact-details-card/contact-details-card';
+import { configureTestSuite } from 'ng-bullet';
+
+describe('ViewTestResultCatHomeTestPage', () => {
+  let fixture: ComponentFixture<ViewTestResultCatHomeTestPage>;
+  let component: ViewTestResultCatHomeTestPage;
+  let loadingController: LoadingController;
+
+  configureTestSuite(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        ViewTestResultCatHomeTestPage,
+        MockComponent(TestDetailsCardComponent),
+        MockComponent(RekeyDetailsCardComponent),
+        MockComponent(RekeyReasonCardComponent),
+        MockComponent(ExaminerDetailsCardComponent),
+        MockComponent(VehicleDetailsCardComponent),
+        MockComponent(TestSummaryCardComponent),
+        MockComponent(ViewTestHeaderComponent),
+        MockComponent(DebriefCardComponent),
+        MockComponent(ErrorMessageComponent),
+        MockComponent(BusinessDetailsCardComponent),
+        MockComponent(ContactDetailsCardComponent),
+      ],
+      imports: [IonicModule, AppModule],
+      providers: [
+        {
+          provide: NavController,
+          useFactory: () => NavControllerMock.instance(),
+        },
+        { provide: NavParams, useFactory: () => NavParamsMock.instance() },
+        { provide: Platform, useFactory: () => PlatformMock.instance() },
+        {
+          provide: LoadingController,
+          useFactory: () => LoadingControllerMock.instance(),
+        },
+        {
+          provide: AuthenticationProvider,
+          useClass: AuthenticationProviderMock,
+        },
+        { provide: SearchProvider, useClass: SearchProviderMock },
+        { provide: CompressionProvider, useClass: CompressionProviderMock },
+      ],
+    });
+  });
+
+  beforeEach(async(() => {
+    fixture = TestBed.createComponent(ViewTestResultCatHomeTestPage);
+    component = fixture.componentInstance;
+    loadingController = TestBed.get(LoadingController);
+  }));
+
+  describe('Class', () => {
+    describe('handleLoadingUI', () => {
+      it('should setup a loading spinner when isLoading is set to true', () => {
+        component.handleLoadingUI(true);
+
+        expect(component.isLoading).toEqual(true);
+        expect(component.loadingSpinner).not.toBeNull;
+      });
+      it('should remove the loading spinner when isLoading is set to false', () => {
+        component.loadingSpinner = loadingController.create();
+
+        component.handleLoadingUI(false);
+
+        expect(component.isLoading).toEqual(false);
+        expect(component.loadingSpinner).toBeNull();
+      });
+    });
+    describe('getTestDetails', () => {
+      it('should correctly generate the data', () => {
+        // TODO - Cat Home , use correct mock
```

Comment:
Soo.... do we need a proper mock here or not ? :D // + other instance of the same todo below